Sources/JWTKit/EdDSA/EdDSAKey.swift

```diff
@@ -0,0 +1,51 @@
+import Foundation
+import Crypto
+
+public struct EdDSAKey {
```

Comment:
It's a bit weird that this can be public _or_ private depending on how it's constructed.

Reply:
From https://www.rfc-editor.org/rfc/rfc8037.html#appendix-A I understood, that this can be the case depending on the use case (signing and verifying or verifying). In our specific use case, we provide a list of keys via an endpoint that does not contain the private key.

Do you think its beneficial to have two different vital types for more clarity?

Reply:
I tend to think so, yes: it prevents you from attempting to do something with the wrong type of key.

Reply:
I changed the implementation. It's explicitly now if you create a private or public key. https://github.com/vapor/jwt-kit/pull/82/commits/12f9a890c5a5829b10c1ff262c128288e745e5cb